### PR TITLE
[ci] Convert ci_nightly.yml to multi-arch style

### PR DESCRIPTION
**Pull request by geomin12**
Converting ci_nightly.yml to multi-arch

Test ran here: https://github.com/ROCm/TheRock/actions/runs/24106521430 (the failure is due to test_labels bug, but #4383 fixes that)

Adding `ci:skip` as this only impacts ci_nightly.yml

### CODE REVIEW

File: .github/workflows/ci_nightly.yml [hunk 1]
```diff
@@ -56,80 +54,63 @@ concurrency:
...
-        variant: ${{ fromJSON(needs.setup.outputs.linux_variants) }}
-    uses: ./.github/workflows/ci_linux.yml
+    uses: ./.github/workflows/multi_arch_ci_linux.yml
     secrets: inherit
     with:
-      amdgpu_families: ${{ matrix.variant.family }}
-      artifact_group: ${{ matrix.variant.artifact_group }}
-      test_runs_on: ${{ matrix.variant.test-runs-on }}
-      benchmark_runs_on: ${{ matrix.variant.benchmark-runs-on }}
-      run_functional_tests: ${{ needs.setup.outputs.run_functional_tests == 'true' }}
```
Comment (ScottTodd) [inline .github/workflows/ci_nightly.yml:82]: These benchmarks / functional tests that run on a nightly `schedule` are still only supported in `ci_linux.yml` and not `multi_arch_ci_linux.yml`, so doing this switch would stall the work that @lajagapp is doing on https://github.com/ROCm/TheRock/pull/3986

Can we try converting ci_asan.yml first? I think the feature set there is closer to what runs on `push` and `pull_request` events.

Comment (geomin12) [reply .github/workflows/ci_nightly.yml:82]: i was doing both in parallel :) asan is next and coming soon

**Comment by ScottTodd** [comment]
I think we'll want to rework ci_nightly.yml together with these issues:
* https://github.com/ROCm/TheRock/issues/2281
* https://github.com/ROCm/TheRock/issues/3334

As currently written, once we add tests to the release workflows, we'll be running duplicate builds and tests:

* release_*.yml triggered
  * build artifacts, packages, etc. for all possible targets
  * publish those packages (either before or after tests)
  * (new) test those artifacts, packages, etc.
* ci_nightly.yml triggered
  * (already duplicate work) build artifacts, packages, etc. for all possible targets
  * test those artifacts, packages, etc.

We could restructure ci_nightly to pull artifacts from the latest nightly release build, or just work that trigger directly into the release workflows. Where we currently trigger pytorch/jax/etc. jobs after building rocm artifacts we could also trigger rocm test jobs.

**Comment by jayhawk-commits** [comment]
Is this work still needed?

**Comment by geomin12** [comment]
> Is this work still needed?

yep! now getting around to doing this conversion. although if release pipelines does this, this will be redundant

**Comment by geomin12** [comment]
Closing this PR as release workflows now handle this: https://github.com/ROCm/TheRock/blob/7b7b238e3e0c2e98436ea230b5114a5b6b946abf/.github/workflows/multi_arch_release_linux.yml#L155-L182
